components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/model/impl/SubscriptionDataLoaderImpl.java

```diff
@@ -444,8 +444,11 @@ private String invokeService(String path, String tenantDomain) throws DataLoadin
             do {
                 try {
                     httpResponse = httpClient.execute(method);
+                    if (HttpStatus.SC_OK != httpResponse.getStatusLine().getStatusCode()) {
+                        throw new DataLoadingException("Error while retrieving subscription");
```

Comment:
better to log/throw received error status code.

Reply:
added log with error response details and status code.